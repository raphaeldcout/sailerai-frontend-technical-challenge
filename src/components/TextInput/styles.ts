import styled, { css } from 'styled-components';

export const InputContainer = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: flex-start;
`;

export const StyledInput = styled.textarea`
  all: unset;
  width: 100%;
  min-height: 40px;
  resize: none;
  font-family: inherit;
  font-size: ${({ theme }) => theme.typography.fontSize.sm};
  line-height: ${({ theme }) => theme.typography.fontSize.md};
  color: ${({ theme }) => theme.colors.surface};
  outline: none;
  white-space: pre-wrap;
  overflow-wrap: break-word;
  background-color: transparent;

  ::placeholder {
    color: ${({ theme }) => theme.colors.neutralLowDark};
  }

  ${({ disabled }) => disabled && css`
    cursor: not-allowed;
  `}
`;
