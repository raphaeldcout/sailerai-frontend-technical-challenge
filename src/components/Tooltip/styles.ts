import styled, { DefaultTheme, css } from 'styled-components';

import { TooltipPositionVariants, TooltipStyledProps } from './types';

const tooltipBoxPositionStyles = {
  top: css`
    bottom: calc(100% + 15px);
  `,
  bottom: css`
    top: calc(100% + 15px);
  `,
  left: css`
    right: calc(100% + 15px);
  `,
  right: css`
    left: calc(100% + 15px);
  `,
};

const tooltipBoxAlignmentStyles = {
  top: {
    left: css`
      left: -24px;
    `,
    center: css`
      left: 50%;
      transform: translate(-50%, 0);
    `,
    right: css`
      right: -24px;
    `,
  },
  bottom: {
    left: css`
      left: -24px;
    `,
    center: css`
      left: 50%;
      transform: translate(-50%, 0);
    `,
    right: css`
      right: -24px;
    `,
  },
  left: {
    left: css`
      top: -22px;
    `,
    center: css`
      top: 50%;
      transform: translate(0, -52%);
    `,
    right: css`
      bottom: -18px;
    `,
  },
  right: {
    left: css`
      top: -22px;
    `,
    center: css`
      top: 50%;
      transform: translate(0, -52%);
    `,
    right: css`
      bottom: -18px;
    `,
  },
};

const arrowPositionStyles = {
  top: {
    left: css`
      left: 24px;
      top: 100%;
    `,
    center: css`
      left: 50%;
      top: 100%;
    `,
    right: css`
      left: calc(100% - 24px);
      top: 100%;
    `,
  },
  bottom: {
    left: css`
      left: 24px;
      bottom: 100%;
    `,
    center: css`
      bottom: 100%;
      left: 50%;
    `,
    right: css`
      bottom: 100%;
      left: calc(100% - 24px);
    `,
  },
  left: {
    left: css`
      right: -20px;
      top: 10px;
    `,
    center: css`
      right: -20px;
      top: 50%;
      transform: translate(0, -50%);
    `,
    right: css`
      right: -20px;
      bottom: 10px;
    `,
  },
  right: {
    left: css`
      left: -10px;
      top: 10px;
    `,
    center: css`
      left: -10px;
      top: 50%;
      transform: translate(0, -50%);
    `,
    right: css`
      left: -10px;
      bottom: 10px;
    `,
  },
};

const getTooltipArrowStyle = (
  tooltipPosition: TooltipPositionVariants,
  theme: DefaultTheme,
) => {
  switch (tooltipPosition) {
    case 'top':
      return `${theme.colors.primaryLight} transparent transparent transparent`;
    case 'bottom':
      return `transparent transparent ${theme.colors.primaryLight} transparent`;
    case 'left':
      return `transparent transparent  transparent ${theme.colors.primaryLight}`;
    case 'right':
      return `transparent ${theme.colors.primaryLight} transparent transparent`;
    default:
      return '';
  }
};

export const TooltipStyled = styled.div<TooltipStyledProps>`
  padding: 10px 16px;
  background-color: ${({ theme }) => theme.colors.primaryLight};
  opacity: 0.8;
  color: ${({ theme }) => theme.colors.surface};
  border-radius: ${({ theme }) => theme.borders.radius.sm};
  position: absolute;
  width: max-content;
  max-width: 270px;
  ${({ tooltipPosition }) => tooltipBoxPositionStyles[tooltipPosition]};
  ${({ tooltipArrowPosition, tooltipPosition }) =>
    tooltipBoxAlignmentStyles[tooltipPosition][tooltipArrowPosition]};
  transition: all 0.5s;

  & > p {
    word-break: break-word;
  }

  &::after {
    content: ' ';
    position: absolute;
    ${({ tooltipArrowPosition, tooltipPosition }) =>
      arrowPositionStyles[tooltipPosition][tooltipArrowPosition]};
    margin-left: -8px;
    border-width: 10px;
    border-style: solid;
    border-color: ${({ theme, tooltipPosition }) =>
      getTooltipArrowStyle(tooltipPosition, theme)};
  }
`;

export const TooltipContent = styled.div`
  display: inline-block;
  position: relative;
`;
