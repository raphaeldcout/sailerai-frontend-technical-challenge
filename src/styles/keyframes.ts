import { keyframes } from 'styled-components';

export const fadeIn = keyframes`
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
`;

export const fadeUp = keyframes`
  from {
    opacity: 0;
    transform: translateY(20px);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
`;

export const typingDots = keyframes`
  0% {
    content: '';
  }
  20% {
    content: '.';
  }
  50% {
    content: '..';
  }
  80% {
    content: '...';
  }
`;

export const waveAnimation = keyframes`
  0%, 100% {
    transform: scaleY(1);
  }
  50% {
    transform: scaleY(1.5);
  }
`;

export const pulse = keyframes`
  0% {
    transform: scale(1);
    box-shadow: 0 0 0 rgba(46, 178, 110, 0.2);
  }
  50% {
    transform: scale(1.05);
    box-shadow: 0 0 15px rgba(46, 178, 110, 0.3);
  }
  100% {
    transform: scale(1);
    box-shadow: 0 0 0 rgba(46, 178, 110, 0.2);
  }
`;

export const slideUpAndFade = keyframes`
  from {
    opacity: 0;
    transform: translateY(2px);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
`;

export const slideRightAndFade = keyframes`
  from {
    opacity: 0;
    transform: translateX(-2px);
  }
  to {
    opacity: 1;
    transform: translateX(0);
  }
`;

export const slideDownAndFade = keyframes`
  from {
    opacity: 0;
    transform: translateY(-2px);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
`;

export const slideLeftAndFade = keyframes`
  from {
    opacity: 0;
    transform: translateX(2px);
  }
  to {
    opacity: 1;
    transform: translateX(0);
  }
`;

export const rotate = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;