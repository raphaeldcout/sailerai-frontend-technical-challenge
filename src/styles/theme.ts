const theme = {
    colors: {
        primary: '#183145',
        primaryLight: '#788AA3',
        primaryDark: '#212121',
        primaryMedium: '#11283a',
        secondary: '#38b3ff',
        background: '#1f1f22',
        surface: '#f5f5f5',
        error: '#d32f2f',
        success: '#388e3c',
        successLight: '#2DD881',
        warning: '#fbc02d',
        border: '#e0e0e0',
        light: '#ffffff',
        neutralLowLight: '#b0d4ee',
        neutralLowMedium: '#f1f1f1',
        neutralLowDark: '#c4c4c4',
        neutralMediumLight: '#6c757d',
        neutralMedium: '#495057',
        greenLowLight: '#dcf6c6',
        rbg: {
            greenLowMedium: 'rgba(46, 178, 110, 0.2)',
            greenLowDark: 'rgba(46, 178, 110, 0.3)',
        }
    },
    darkMode: {
        foregroundRgb: '255, 255, 255',
        backgroundStartRgb: '0, 0, 0',
        backgroundEndRgb: '19, 19, 22',
    },
    spacings: {
        xs: '4px',
        sm: '8px',
        md: '16px',
        lg: '24px',
        xl: '32px',
        xxl: '64px',
    },
    typography: {
        fontSize: {
            xs: '12px',
            sm: '14px',
            md: '16px',
            lg: '20px',
            xl: '24px',
            xxl: '32px',
        },
        fontWeight: {
            light: 300,
            normal: 400,
            medium: 500,
            bold: 700,
        },
        lineHeight: {
            xs: '1rem',
            sm: '1.2rem',
            md: '1.4rem',
            lg: '1.6rem',
            xl: '1.8rem',
        },
    },
    borders: {
        radius: {
            none: '0px',
            sm: '4px',
            md: '8px',
            lg: '12px',
            full: '999px',
        },
        width: {
            thin: '1px',
            thick: '2px',
        },
    },
    shadows: {
        sm: '0px 1px 3px rgba(0, 0, 0, 0.12)',
        md: '0px 3px 6px rgba(0, 0, 0, 0.16)',
        lg: '0px 10px 20px rgba(0, 0, 0, 0.2)'
    },
    zIndex: {
        dropdown: 1000,
        modal: 1100,
        tooltip: 1200,
    },
    breakpoints: {
        xsmall: 575.98,
        small: 767.98,
        medium: 991.98,
        large: 1199.98,
        xlarge: 1399.98
    }
};

export const colors = theme.colors;

export type Theme = typeof theme;

export type ColorsType = keyof Theme['colors'];

export default theme;