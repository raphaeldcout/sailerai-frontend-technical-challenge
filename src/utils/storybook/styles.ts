import styled, { css } from 'styled-components';

import { FlexboxProps } from './types';

export const Flexbox = styled.div<FlexboxProps>`
    display: flex;
    ${({ flexDirection }) => flexDirection && css({ flexDirection })};
    ${({ flexWrap }) => flexWrap && css({ flexWrap })};
    ${({ justifyContent }) => justifyContent && css({ justifyContent })};
    ${({ alignItems }) => alignItems && css({ alignItems })};
    ${({ alignContent }) => alignContent && css({ alignContent })};
    ${({ margin }) => margin && css({ margin })};
    ${({ padding }) => padding && css({ padding })};
    ${({ flex }) => flex && css({ flex })};
    ${({ gap }) => gap && css({ gap })};
    ${({ maxWidth }) => maxWidth && css({ maxWidth })};
  `;