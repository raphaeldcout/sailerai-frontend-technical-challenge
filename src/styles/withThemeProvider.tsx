'use client';

import { FC, PropsWithChildren } from 'react';

import { ThemeProvider } from 'styled-components';

import StyledComponentsRegistry from '@/lib/registry';
import theme from '@/styles/theme';

export const ThemeProviderSailer: FC<PropsWithChildren> = ({ children }) => {
  return (
    <StyledComponentsRegistry>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </StyledComponentsRegistry>
  );
};
