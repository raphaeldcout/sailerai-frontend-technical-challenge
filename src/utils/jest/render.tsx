import { FC, ReactElement, ReactNode } from 'react';

import { ThemeProviderSailer } from '@/styles/withThemeProvider';
import { render, RenderOptions } from '@testing-library/react';

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <ThemeProviderSailer>{children}</ThemeProviderSailer>
);

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };

