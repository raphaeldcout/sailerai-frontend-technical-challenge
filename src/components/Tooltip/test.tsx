import '@testing-library/jest-dom';

import { FiEdit } from 'react-icons/fi';

import theme from '@/styles/theme';
import { render } from '@/utils/jest/render';

import { Tooltip } from '.';
import { TooltipContent } from './styles';

describe('Tooltip', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <Tooltip text="Lorem ipsum is typically">
        <TooltipContent>
          <FiEdit size={20} color={theme.colors.primaryMedium} />
        </TooltipContent>
      </Tooltip>
    );

    expect(container).toMatchSnapshot();
  });
});
