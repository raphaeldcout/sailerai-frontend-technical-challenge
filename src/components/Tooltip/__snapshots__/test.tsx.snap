// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should match snapshot 1`] = `
<div>
  <div
    class="sc-f90ad502-1 UMuxG"
    data-state="closed"
  >
    <svg
      color="#11283a"
      fill="none"
      height="20"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      style="color: rgb(17, 40, 58);"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
      />
      <path
        d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
      />
    </svg>
  </div>
</div>
`;
